runtime/lua/vim/lsp/buf.lua

```diff
@@ -47,6 +47,16 @@ local function request_with_options(name, params, options)
   request(name, params, req_handler)
 end
 
+local function client_request_with_options(client, name, params, options, bufnr)
```

Comment:
don't repeat parameters in the name, unless it is truly necessary to disambiguate from some other local function. the docstring should be enough (which this currently lacks)

```suggestion
--- Calls client.request with the given `opts`.
local function client_request(client, name, params, opts, bufnr)
```